lib/ansible/modules/network/nxos/nxos_interface_ospf.py

```diff
@@ -394,7 +394,16 @@ def main():
                                                'message_digest_password']],
                            supports_check_mode=True)
 
-    if not module.params['interface'].startswith('loopback') and not module.params['interface'].startswith('port-channel'):
+    # Normalize interface input data.
+    #
+    # * For port-channel and loopback interfaces expection is all lower case names.
+    # * All other interfaces the expectation is an uppercase leading character
+    #   followed by lower case characters.
+    #
+    if re.match(r'(port-channel|loopback)', module.params['interface'], re.I):
+        module.params['interface'] = module.params['interface'].lower()
+    else:
+        module.params['interface'] = module.params['interface'].lower()
```

Comment:
`string.capitalize()` ensures the first letter is uppercase and the rest are lowercase, I can't see any reason to keep this line here.

Reply:
Your right. I will update and remove it.